wolfssl/ssl.h

```diff
@@ -1753,15 +1753,15 @@ WOLFSSL_API int wolfSSL_BIO_get_md_ctx(WOLFSSL_BIO *bio,
 WOLFSSL_API WOLFSSL_BIO_METHOD* wolfSSL_BIO_f_buffer(void);
 WOLFSSL_API long wolfSSL_BIO_set_write_buffer_size(WOLFSSL_BIO* bio, long size);
 WOLFSSL_API WOLFSSL_BIO_METHOD* wolfSSL_BIO_f_ssl(void);
-WOLFSSL_API WOLFSSL_BIO*        wolfSSL_BIO_new_socket(int sfd, int flag);
-WOLFSSL_API WOLFSSL_BIO*        wolfSSL_BIO_new_dgram(int fd, int closeF);
+WOLFSSL_API WOLFSSL_BIO*        wolfSSL_BIO_new_socket(SOCKET_T sfd, int flag);
```

Comment:
Changing public API. And it doesn't match OpenSSL. (Yes OpenSSL is wrong.)
Make a new one that is SOCKET_T.

Reply:
not bothering with the new APIs for now -- may not in practice ever be needed.

in any case I've reverted all the changes to public APIs.  now, all substantive header changes in the PR are in `internal.h`.